### PR TITLE
Transition to Jest, Increase Languages Tested, and Fix Bug with Sub-Languages

### PR DESCRIPTION
**Pull request by keplersj**

### CODE REVIEW

**Comment by felixfbecker** [comment]
Why the change to jest?

**Comment by keplersj** [comment]
Primarily to make use of Snapshots.

**Comment by felixfbecker** [comment]
Could you explain that feature or link me to docs? I am always a bit reluctant to switch to different frameworks without evaluating the pros/cons.

**Comment by keplersj** [comment]
No worries. Totally get it. Currently this repo takes an input and tracks it against an expected output. Snapshots simply automate the tracking of the expected output by placing it in external snapshot files. This is super helpful for dynamically generated tests, which this PR also does by looping through fixtures and testing against them.

For more on Snapshot testing here's a [talk from React Conf 2017](https://www.youtube.com/watch?time_continue=416&v=HAuXJVI_bUs) about it, and the [documentation on Snapshot Testing](https://facebook.github.io/jest/docs/en/snapshot-testing.html#content)

**Comment by felixfbecker** [comment]
I generally like to compose unopinionated libraries, but this sounds like a useful feature that would require boilerplate to replicate with mocha.

File: .travis.yml [hunk 1]
```diff
@@ -16,11 +16,9 @@ install:
 
 script:
   - npm run lint
-  - npm run build
```
Comment (felixfbecker) [inline .travis.yml]: I  would still like to run this in CI to see if the compilation has any errors.
Then of course the compilation that jest does is a bit redundant, is it possible to just run jet from the output JS?

File: src/test/test.ts [hunk 2]
```diff
@@ -1,94 +1,27 @@
...
-function test(language: string, code: string, expected: string): void {
+function test(language: string, code: string): void {
     it(`should color ${language} correctly`, () => {
         const highlighted = highlight(code)
-        if (highlighted.trim() !== expected.trim()) {
-            throw new HighlightAssertionError(highlighted, expected)
-        } else {
-            console.log(highlighted + '\n')
+
+        if (process.env.VERBOSE) {
```
Comment (felixfbecker) [inline src/test/test.ts]: Can you set this in CI? In opposite to the previous assertion way, the snapshot files contain the raw ANSI codes, which is impossible to code review. Printing the coloured output in CI is a good and easy way to verify that the changes the snapshots were correct.

File: .travis.yml [hunk 3]
```diff
@@ -16,11 +16,9 @@ install:
...
-  - npm run cover
+  - npm run test
 
 after_success:
-  - nyc report --reporter=json
```
Comment (felixfbecker) [inline .travis.yml]: jest seems to always generate JSON, lcov and clover format, which are then _all three_ uploaded to codecov. I am a bit worried that if one format is better than the other (does lcov support sub-line coverage?) we get the lowest common denominator of features as codecov merges the coverage. Is there a way to only tell jest to generate JSON?

File: .travis.yml (hunk 1 above)
Comment (keplersj) [reply .travis.yml]: To work with TypeScript I set up Jest to use a package called [`ts-jest`](https://github.com/kulshekhar/ts-jest). Running tsc before Jest wouldn't have any effect on the tests. Happy to add it back

File: .travis.yml (hunk 3 above)
Comment (keplersj) [reply .travis.yml]: Can do using this [configuration option](https://facebook.github.io/jest/docs/en/configuration.html#coveragereporters-array-string)

File: src/test/test.ts (hunk 2 above)
Comment (keplersj) [reply src/test/test.ts]: Of course!

File: .travis.yml (hunk 1 above)
Comment (felixfbecker) [reply .travis.yml]: So it's not possible to just run the generated .js files with jest?

File: .travis.yml (hunk 3 above)
Comment (felixfbecker) [reply .travis.yml]: Cool, lets set that to `text-summary` and `json`

File: src/test/test.ts (hunk 2 above)
Comment (keplersj) [reply src/test/test.ts]: Printing code samples on CI now https://travis-ci.org/felixfbecker/cli-highlight/builds/306902109

File: .travis.yml (hunk 3 above)
Comment (keplersj) [reply .travis.yml]: That does look better https://codecov.io/gh/felixfbecker/cli-highlight/pull/10/src/src/index.ts?before=src/index.ts

File: .travis.yml (hunk 1 above)
Comment (keplersj) [reply .travis.yml]: It's possible. Using ts-jest ensures that code coverage reports produced are accurate. But it's not necessary https://github.com/facebook/jest/issues/336#issuecomment-325804591 https://github.com/codecov/example-typescript/issues/7

Comment (felixfbecker) [reply .travis.yml]: Okay, that's a priority of course. Compilation should be fast so not worried that much.

Comment (keplersj) [reply .travis.yml]: So what were you thinking on this one? Stick with `ts-jest` or, compile then test like in the SourceGraph example?

Comment (felixfbecker) [reply .travis.yml]: We can use ts-jest. What do you mean with SourceGraph example?

Comment (keplersj) [reply .travis.yml]: Err. My bad. The Codecov example. https://github.com/codecov/example-typescript

**Comment by keplersj** [comment]
Woot! 🎉 Thank you!

**Comment by felixfbecker** [comment]
Thanks for your contributions! You turned this into a really solid module

**Pull request merged**